File: ext/lua/examples/multi.lua
-- ext/lua/examples/multi.lua —— **多值**那一批节点的例子（第二个例子家族）
--
-- 期望输出（与 ext/go/examples/multi.go 逐行相同）：
--   3
--   7
--   1 2
--
-- 三行各压一格：
--   `return a, b`        -> values（多值的生产侧 —— **多出端口是常态**）
--   `local lo, hi = f()` -> 一格临时 bind + 一串 pick（消费侧，五门语言共用 destructure）
--   `print(f())`         -> 实参表里**只有最后一格展开**（arity 契约那一条）

local function minmax(a, b)
  if a < b then
    return a, b
  end
  return b, a
end

local lo, hi = minmax(7, 3)
print(lo)
print(hi)
print(minmax(1, 2))

File: ext/go/examples/multi.go
// ext/go/examples/multi.go —— 与 ext/lua/examples/multi.lua **同一件事**
//
// 期望输出逐行相同：3 / 7 / 1 2。
// Go 这一份多压一格 lua 没有的：**多值直接当实参**（`fmt.Println(minmax(1, 2))`
// 在 Go 里合法），所以"列表里只有最后一格展开"那条 arity 契约两门语言一起验。

package main

import "fmt"

func minmax(a int, b int) (int, int) {
	if a < b {
		return a, b
	}
	return b, a
}

func main() {
	lo, hi := minmax(7, 3)
	fmt.Println(lo)
	fmt.Println(hi)
	fmt.Println(minmax(1, 2))
}
